lib/ood_core/job/adapters/kubernetes/templates/pod.yml.erb

```diff
@@ -20,12 +20,10 @@ spec:
     runAsUser: <%= run_as_user %>
     runAsGroup: <%= run_as_group %>
     runAsNonRoot: true
-    <%- if spec.container.supplemental_groups.empty? -%>
-    supplementalGroups: []
-    <%- else -%>
+    <%- unless spec.container.supplemental_groups.empty? -%>
     supplementalGroups:
     <%- spec.container.supplemental_groups.each do |supplemental_group| -%>
-    - "<%= supplemental_group %>"
+    - <%= supplemental_group %>
     <%- end -%>
```

Comment:
Seems like the template should just be this with no blocks. We're guaranteeing it's empty right?  I'm just concerned with how unmanageable this template is becoming with switches everywhere.

```ruby
supplementalGroups: <%= spec.container.supplemental_groups %>
```
We have test cases where it's empty, but we'd need some test cases around it being populated too. (the test cases too could use a little re-factoring given all the options/ways we initialize the subject).

Reply:
There already is one test case with `enable_supplemental_groups` cluster option being enabled so that covers the default groups getting added.  I added another test case around merging submit.yml group list with default group list.

Want me to refactor tests here or just get tests covering new behavior and refactor in different pull request?

Reply:
Also added commit to simplify the template per your suggestion.

Reply:
just get tests covering new behavior. There's no need to refactor in this PR, I'm just complaining. I just want to be sure that we actually do _guarantee_ an empty array every time.

Reply:
Then that should be covered, the majority of YAML file tests verify default behavior is to do empty supplemental groups, then one test verifies we populate defaults and another test verifies we populate defaults and merge with submit.yml groups.